### DISCUSSION TITLE
Class A and Class C transmission error

### DISCUSSION THREAD

**Discussion by kvkc97** [Lora-net/LoRaMac-node#1170] [Q&A]
I am using periodic-uplink-lpp to send data continuously from a custom board using STM32WB55RG and LR1110. I have tried it using Class A and Class C. 
In Class A, uplink was ok but downlink was not received at the end device(it is forwarded from the gateway). RX1 window is opened correctly but RX2 window is not opened. TX Periodicity is set as 7 sec.
In Class C, the RX windows are correctly opened but uplink was not send and downlink was not received. Class C timeout is set as 8 sec.
For downlink, I checked for the RXDONE interrupt flag to be set but it was not set. What could be issue happening here/

**Comment by mluis1** [comment]
The issue is most certainly related to the port to STM32WB55RG mcu target. I would recommend to verify that the ported files are implemented as expected.

What we can tell is that the examples provided by this project are passing the LoRaWAN certification for Class A and Class C.

Another thing that could cause problems is related to the LR1110 radio as you may have a more recent version of the radio firmware than the one currently supported by this project. It is in our plans to update the LR1110 drivers to the latest version however we cannot provide a time frame when this will happen.
